### PR TITLE
GAWB-805: admin endpoint for google genomics api

### PR DESCRIPTION
**Pull request by davidangb**
I decided to do this the "long way" by including the Google Genomics client library and making a dedicated GenomicsService, in case we end up wanting to call other genomics endpoints.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: If PR includes new or changed db queries, include the explain plans in the description
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+class GenomicsService(protected val userInfo: UserInfo, val dataSource: SlickDataSource, protected val gcsDAO: GoogleServicesDAO, userDirectoryDAO: UserDirectoryDAO)(implicit protected val executionContext: ExecutionContext) extends Actor with AdminSupport with FutureSupport with UserWiths {
+
+  override def receive = {
+    case GetOperation(jobId) => getOperation(jobId) pipeTo sender
+  }
+
+  def getOperation(jobId: String): Future[PerRequestMessage] = {
+    gcsDAO.getGenomicsOperation(jobId).map {
```
Comment (skwalker) [inline src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala]: Should this be within an `asAdmin` block?

**Comment by skwalker** [comment]
There should probably be a test in AdminApiServicSpec for the new endpoint, right?

From what I can tell, everything looks fine -- seems like we're doing things the same way when we're doing similar things in other places in the code.

File: src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala (hunk 1 above)
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala]: oh yes it should, duh! will fix.

File: src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  sealed trait GenomicsServiceMessage
+  case class GetOperation(jobId: String) extends GenomicsServiceMessage
+
+}
+
+class GenomicsService(protected val userInfo: UserInfo, val dataSource: SlickDataSource, protected val gcsDAO: GoogleServicesDAO, userDirectoryDAO: UserDirectoryDAO)(implicit protected val executionContext: ExecutionContext) extends Actor with AdminSupport with FutureSupport with UserWiths {
+
+  override def receive = {
+    case GetOperation(jobId) => getOperation(jobId) pipeTo sender
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/genomics/GenomicsService.scala]: put the asAdmin up here

**Comment by dvoet** [comment]
agreed it would be nice to have some tests but probably in integration test land. I don't think we should launch a real job though so this may be difficult.

**Comment by dvoet** [comment]
If tests prove to difficult I am ok to let it slide, otherwise 👍

**Comment by davidangb** [comment]
@dvoet @skwalker I've updated to include asAdmin (_blush_). I've added a rudimentary test that validates the spray routing and calls into the mock version of the GoogleServicesDAO, but I don't think we can go too much further than that.

**Comment by skwalker** [comment]
lgtm!

File: build.sbt [hunk 3]
```diff
@@ -45,6 +45,7 @@ libraryDependencies ++= {
     ("com.google.apis" % "google-api-services-plus" % "v1-rev381-1.20.0"),
     ("com.google.apis" % "google-api-services-admin-directory" % "directory_v1-rev53-1.20.0"),
     "com.google.apis" % "google-api-services-oauth2" % "v1-rev112-1.20.0",
+    "com.google.apis" % "google-api-services-genomics" % "v1-rev88-1.22.0",
```
Comment (pgrosu) [inline build.sbt]: I would update this to `v1-rev89-1.22.0` according to the latest available version here:

https://oss.sonatype.org/#nexus-search;quick~google-api-services-genomics

Comment (davidangb) [inline build.sbt]: thank you!

Comment (pgrosu) [inline build.sbt]: You're welcome :)

**Pull request merged**